test/integration/shim_int_test.go

```diff
@@ -106,16 +116,41 @@ name = input("Enter your name: ")
 print("Hello {}!".format(name))
 `
 
-	testScript := filepath.Join(fmt.Sprintf("%s/%s.pl", ts.Dirs.Work, suite.T().Name()))
+	testScript := filepath.Join(fmt.Sprintf("%s/%s.py", ts.Dirs.Work, suite.T().Name()))
 	fail := fileutils.WriteFile(testScript, []byte(scriptBlock))
 	suite.Require().NoError(fail.ToError())
 
-	cp := ts.Spawn("shim", "--", "python3", fmt.Sprintf("%s", testScript))
+	cp := ts.SpawnWithOpts(
+		e2e.WithArgs("shim", "--", "python3", fmt.Sprintf("%s", testScript)),
+		e2e.AppendEnv("ACTIVESTATE_CLI_DISABLE_RUNTIME=false"),
+	)
 	cp.SendLine("ActiveState")
 	cp.Expect("Hello ActiveState!")
 	cp.ExpectExitCode(0)
 }
 
+func (suite *ShimIntegrationTestSuite) TestShim_SystemPython() {
+	_, err := exec.LookPath("python3")
+	if err != nil {
+		suite.T().Skip("Cannot run test if system does not have python installation")
+	}
+
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	scriptBlock := `
+print("Hello World!")
+`
+
+	testScript := filepath.Join(fmt.Sprintf("%s/%s.py", ts.Dirs.Work, suite.T().Name()))
+	fail := fileutils.WriteFile(testScript, []byte(scriptBlock))
+	suite.Require().NoError(fail.ToError())
+
+	cp := ts.Spawn("shim", "--", "python3", testScript)
+	cp.Expect("Hello World!")
+	cp.ExpectExitCode(0)
+}
+
 func TestShimIntegrationTestSuite(t *testing.T) {
 	suite.Run(t, new(ShimIntegrationTestSuite))
 }
```

Comment:
Can we add an integration test that tests the variant without `--`. eg. `state shim python3 foo.py`